qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -208,23 +238,50 @@ def __init__(self, squadrons: list[Squadron], theater: ConflictTheater) -> None:
 
         layout.addWidget(scrolling_area)
 
-    def apply(self) -> list[Squadron]:
-        return self.squadrons_config.apply()
+    def addSquadron(self, squadron: Squadron) -> None:
+        self.squadrons_config.addSquadron(squadron)
+
+    def removeSquadron(self, squadron: Squadron) -> None:
+        self.squadrons_config.removeSquadron(squadron)
 
 
 class AircraftSquadronsPanel(QStackedLayout):
     def __init__(self, air_wing: AirWing, theater: ConflictTheater) -> None:
         super().__init__()
         self.air_wing = air_wing
+        self.theater = theater
+        self.air_wing.subscribe(self.handleChanges)
+
         self.squadrons_pages: dict[AircraftType, AircraftSquadronsPage] = {}
         for aircraft, squadrons in self.air_wing.squadrons.items():
-            page = AircraftSquadronsPage(squadrons, theater)
+            page = AircraftSquadronsPage(squadrons, self.theater, self.air_wing)
             self.addWidget(page)
             self.squadrons_pages[aircraft] = page
 
-    def apply(self) -> None:
-        for aircraft, page in self.squadrons_pages.items():
-            self.air_wing.squadrons[aircraft] = page.apply()
+    def __del__(self) -> None:
+        self.air_wing.unsubscribe(self.handleChanges)
+
+    def handleChanges(self, event) -> None:
```

Comment:
Add a `Signal` per event.

```python
aircraft_type_added = Signal(AircraftType)
aircraft_type_removed = Signal(AircraftType)
squadron_added = Signal(Squadron)
squadron_removed = Signal(Squadron)
```

Connect those to the various update methods you have, and emit those as needed.